File: 9-palindrome-number/9-palindrome-number.ts
function isPalindrome(x: number): boolean {
    let temp:number = x;
    let sum:number = 0;
    
    if(Math.sign(x) === -1){
        return false;
    }
    while(x != 0){
        let rem:number = x % 10;
        sum = sum * 10 + rem;
        x = Math.floor(x / 10);
    }
    return temp === sum;
};

File: 9-palindrome-number/9-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int temp = x;
        int sum = 0;
        if(Math.signum(x) == -1){
            return false;
        }
        while(x != 0){
            int rem = x % 10;
            sum = sum * 10 + rem;
            x = x / 10;
        }
        return temp == sum;
    }
}